main.py

```diff
@@ -311,33 +321,43 @@ def command_unknown_user(self, update: Update, context: CallbackContext) -> None
         self.send_text(update, context, self.telegram.rejected_message)
 
     def command_turn_on(self, update: Update, context: CallbackContext) -> None:
-        self.device_actuator(update, context, "on", "Turned on {}.")
+        self.device_actuator(update, context, "on", "/on", "Turned on {}.")
 
     def command_turn_off(self, update: Update, context: CallbackContext) -> None:
-        self.device_actuator(update, context, "off", "Turned off {}.")
+        self.device_actuator(update, context, "off", "/off", "Turned off {}.")
 
     def command_list_users(self, update: Update, context: CallbackContext) -> None:
         self.request_access(update, context, AccessLevel.ADMIN)
 
         def row(id, isAdmin, userGroup, deviceGroup) -> str:
             return f"{id :10}|{isAdmin :5}|{userGroup :10}|{deviceGroup}"
 
-        text = ["```", row("Id", "Admin", "UserGroup", "DeviceGroups"), "----------|-----|----------|-----------"]
-        text += [row(u.id, str(u.is_admin), u.user_group, [group.name for group in u.device_groups]) for u in self.telegram.users.values()]
+        text = ["```", row("Id", "Level", "UserGroup", "DeviceGroups"), "----------|-----|----------|-----------"]
+        text += [row(u.id, u.access_level, u.user_group, [group.name for group in u.device_groups]) for u in self.telegram.users.values()]
         text.append("```")
         self.send_md(update, context, text)
 
+    def get_percent(self, input: str) -> int:
+        percent = -1
+        try:
+            percent = int(input)
+        except ValueError:
+            return None
+        if percent < 0 or percent > 100:
+            return None
+        return percent
+
     def command_dim(self,  update: Update, context: CallbackContext) -> None:
         self.request_access(update, context, AccessLevel.DEVICE)
         if len(context.args) < 2:
-            self.send_text("Dim level and device name must be specified.")
+            self.send_text(update, context, "Dim level and device name must be specified.")
```

Comment:
is this assumption correct that any 2 parameters specified are have to be level and device name?  What if there are 3 params?

Reply:
the bot library separates cmd args with spaces.
therefore,
/dim 20 downstairs office light
will show as 4 args.
so what I do is take the first arg, then " ".join() the rest.
Alternative would be to force the user to quote device names which is more typing and a pain in the neck on phones.